packages/experimental/Badge/src/CounterBadge/useCounterBadge.ts

```diff
@@ -0,0 +1,21 @@
+import { CounterBadgeProps, CounterBadgeInfo } from './CounterBadge.types';
+
+export const useCounterBadge = (props: CounterBadgeProps): CounterBadgeInfo => {
+  const DEFAULT_OVERFLOW_COUNT = 99;
+  const { dot, overflowCount = DEFAULT_OVERFLOW_COUNT, showZero, count, ...rest } = props;
+
+  const _count = count && count > overflowCount ? `${overflowCount}+` : `${count}`;
```

Comment:
nit: I think it'd be better to use a different convention to denote a local variable that clashes with a prop name, since the _ usually means "unused" in a JS context.

I've seen things like "countInner" or renaming the prop version to "countOrig"

Reply:
I guess the _ also means "private" though. Not a blocker, just an opinion/

Reply:
Yeah, I thought about using it as private. But I'll change to countInner